picasso/src/main/java/com/squareup/picasso3/BitmapHunter.java

```diff
@@ -199,24 +190,16 @@ Result hunt() throws IOException {
       Bitmap bitmap = result.getBitmap();
       stats.dispatchBitmapDecoded(bitmap);
 
-      int exifOrientation = result.getExifOrientation();
-      if (data.needsTransformation() || exifOrientation != 0) {
-        if (data.needsMatrixTransform() || exifOrientation != 0) {
-          bitmap = transformResult(data, bitmap, exifOrientation);
-          if (picasso.loggingEnabled) {
-            log(OWNER_HUNTER, VERB_TRANSFORMED, data.logId());
-          }
-        }
+      List<Transformation> transformations = new ArrayList<>();
 
-        result = new Result(bitmap, result.getLoadedFrom(), exifOrientation);
-        if (data.hasCustomTransformations()) {
-          result = applyCustomTransformations(data.transformations, result);
-          if (picasso.loggingEnabled) {
-            log(OWNER_HUNTER, VERB_TRANSFORMED, data.logId(),
-                "from custom transformations");
-          }
-        }
+      if (data.needsMatrixTransform() || result.getExifOrientation() != 0) {
+        transformations.add(new MatrixTransformation(data));
+      }
+      if (data.hasCustomTransformations()) {
```

Comment:
Let's delete this method and normalize to `emptyList()` in the constructor (and back to null when calling `newBuilder`). This way we never need to check and can always call `addAll`.

Reply:
Can be follow-up as well.

Reply:
Updated.